notes/2024/2024-07-25.md

```diff
@@ -0,0 +1,88 @@
+
+### Statuses
+
+* **@nzakas:**  has been working on `@eslint/json`, `@eslint/markdown`, the new config loading RFC, and started work on the core rewrite RFC (but that will take a while).
+
+* **@mdjermanovic:** finished `eslint-plugin-react` support for ESLint v9, and was mostly reviewing PRs.
+
+* **@fasttime:** finished updating `@types/eslint` to v9 and I've been also mostly busy reviewing issues and PRs.
+
+
+### [feat: merge `rule.meta.defaultOptions` before validation](https://github.com/eslint/eslintrc/pull/166)
+
+The Code Inspector code has been pushed to GitHub and we have a dev version running. There are definitely some rough edges still to sand down.
+
+**Action Items:** @fasttime will do a review as well
+
+### [Change Request: Better support migrating JS config files](https://github.com/eslint/rewrite/issues/64)
+
+**Action Items:** @fasttime will look into this after looking at the Code Inspector PR.
+
+### [feat!: Look Up Config Files From Linted File](https://github.com/eslint/rfcs/pull/120)
+
+Regarding [this comment](https://github.com/eslint/rfcs/pull/120#issuecomment-2251117016) by @mdjermanovic:
+
+> I agree with option 2 as presented in [#120 (comment)](https://github.com/eslint/rfcs/pull/120#issuecomment-2221153450): with that setup, `eslint` should not traverse into `subir`. But, with the same setup, I'm still not sure if `eslint subdir`, and especially `eslint subdir/foo.js` should result in files being ignored (as in the FAQ of this RFC document).
+
+Meeting thread:
+
+> @nzakas: I think the distinction you're making is between explicitly passing the directory/file vs. having it match a pattern, correct?
+
+> @mdjermanovic: Mostly, but it could be with patterns too. `eslintrc`, I believe, starts looking for config files from the pattern "root" (that's part of the pattern that represents a literal path, what `glob-parent` returns). I think ignoring files/directories is the only open question, everything else is pretty clear, so my suggestion is to leave that part of the design for discussing on the implementation
+
+> @nzakas: If we agree that `eslint .` should ignore `subdir`, then that means we need to honor `./eslint.config.js`. And if we honor it, then it seems like we should honor it 100% of the time. The bottom line is that in `eslintrc`, you always get the same result regardless of how you pass the file path on the command line, and I think flat config should do the same. So if `eslint .` ignores `subdir` then so should `eslint subdir` and `eslint subdir/foo.js`. If `eslint .` does not ignore `subdir` then the same should go for `eslint subdir` and `eslint subdir/foo.js`. I don't think mixing and matching makes any sense
+
+> @mdjermanovic: in this [stackblitz](https://discord.com/channels/688543509199716507/688545247843713092/1266133982689628303) example `eslint .` doesn't go into `subdir` because of the ignorePattern in the cwd config file, but `eslint subdir` and `eslint subdir/file.js` do lint the file. 
```

Comment:
```suggestion
> @mdjermanovic: in this [stackblitz](https://stackblitz.com/edit/stackblitz-starters-wjz9em?file=.eslintrc.json,package.json,subdir%2F.eslintrc.json) example `eslint .` doesn't go into `subdir` because of the ignorePattern in the cwd config file, but `eslint subdir` and `eslint subdir/file.js` do lint the file. 
```